### PR TITLE
feat: reactive remote management

### LINKED ISSUE #1365: Make remote management reactive
When adding remotes and they are found the list should update without a refresh.

### PR DESCRIPTION
**Pull request by xla**
Closes #1365

### CODE REVIEW

File: proxy/coco/src/peer/run_state.rs [hunk 1]
```diff
@@ -67,9 +78,26 @@ impl MaybeFrom<&Input> for Event {
             Input::Announce(input::Announce::Succeeded(updates)) => {
                 Some(Self::Announced(updates.clone()))
             },
-            Input::Peer(event) => Some(Self::Peer(event.clone())),
+            Input::Peer(event) => match event {
+                PeerEvent::GossipFetch(FetchInfo {
```
Comment (FintanH) [inline proxy/coco/src/peer/run_state.rs:82]: Can we provide a `From` impl?

**Review by FintanH — DISMISSED** [review]
Just the one suggestion :+1:

**Review by FintanH — DISMISSED** [review]

File: proxy/coco/src/peer/run_state.rs (hunk 1 above)
Comment (xla) [reply proxy/coco/src/peer/run_state.rs:82]: From which type to which?

Comment (FintanH) [reply proxy/coco/src/peer/run_state.rs:82]: `FetchInfo` to GossipFetched`, or even just re-use `FetchInfo` in the `Event`

**Comment by rudolfs** [comment]
Screenshot 2020-11-27 at 17 50 50

**Review by FintanH — APPROVED** [review]
Works SO WELL for me

**Review by rudolfs — APPROVED** [review]

**Pull request merged**